controllers/imagerepository_controller.go

```diff
@@ -240,6 +260,45 @@ func getAwsECRLoginAuth(accountId, awsEcrRegion string) (authn.AuthConfig, error
 	return authConfig, nil
 }
 
+// getGCRLoginAuth obtains authentication for the image by
+// getting a token from the metadata API on GCP. This assumes that
+// the pod has right to pull the image which would be the case if it
+// is hosted on GCP. It works with both service account and workload identity
+// enabled clusters.
+func getGCRLoginAuth() (authn.AuthConfig, error) {
+	var authConfig authn.AuthConfig
+	gcpDefaultTokenURL := "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token"
+
+	request, err := http.NewRequest("GET", gcpDefaultTokenURL, nil)
```

Comment:
Maybe use the const here
```suggestion
	request, err := http.NewRequest(http.MethodGet, gcpDefaultTokenURL, nil)
```